### PR TITLE
Remove client delaration from compiler and parser

### PR DESCRIPTION
**Pull request by SasinduDilshara**

### CODE REVIEW

**Review by azinneera — APPROVED** [review]

**Comment by azinneera** [comment]
Project API changes LGTM.

**Review by gimantha — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java [hunk 1]
```diff
@@ -1464,7 +1463,8 @@ public void visit(BLangXMLSequenceLiteral xmlSequenceLiteral) {
     @Override
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java:1466]: Is this change related to IDL Client Change?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java [hunk 2]
```diff
@@ -3414,7 +3409,8 @@ public void visit(BLangAnnotAccessExpr annotAccessExpr, AnalyzerData data) {
     @Override
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral, AnalyzerData data) {
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:3412]: Check this also if this is related to IDL client, IMO it is not.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java [hunk 3]
```diff
@@ -2206,7 +2195,8 @@ private void addUninitializedVar(BLangVariable variable) {
     @Override
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
+                        new ArrayList<>());
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java:2199]: Same as above

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java [hunk 4]
```diff
@@ -2010,15 +2009,11 @@ public void visit(BLangXMLNavigationAccess xmlNavigation) {
...
-    @Override
-    public void visit(BLangClientDeclaration clientDeclaration) {
-    }
-
     @Override
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
+                        new ArrayList<>());
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java:2016]: This also

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BNeverType.java [hunk 5]
```diff
@@ -44,9 +43,4 @@ public boolean isNullable() {
     public String toString() {
         return Names.NEVER.value;
     }
-
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BNeverType.java:47]: I think it is ok not to remove this one

File: compiler/ballerina-parser/src/test/resources/declarations/client-decl/client_decl_source_15.bal [hunk 6]
```diff
@@ -1,3 +1,3 @@
 function fn() {
-    client "http://www.example.com/apis/one.yaml" as foo;
+client "http://www.example.com/apis/one.yaml" as foo;
```
Comment (gimantha) [inline compiler/ballerina-parser/src/test/resources/declarations/client-decl/client_decl_source_15.bal:2]: this test is no longer valid

File: language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation1.bal [hunk 7]
```diff
@@ -1,3 +1,14 @@
-import ballerina/module1;
+public type AnnotationType record {
```
Comment (gimantha) [inline language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation1.bal:1]: Please check if this is related to this IDL client feature. I dont think do. https://github.com/ballerina-platform/ballerina-lang/pull/31342

File: language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation2.bal [hunk 8]
```diff
@@ -1,14 +0,0 @@
...
-
-public annotation AnnotationType annotationCommon;
-
-public annotation AnnotationType a1 on service;
-
-public annotation AnnotationType a2 on function;
-
-annotation AnnotationType a3 on service;
-
-@
```
Comment (gimantha) [inline language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation2.bal:14]: This file is not related to IDL client feature. It should not be removed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 9]
```diff
@@ -5845,7 +5829,8 @@ public void visit(BLangStringTemplateLiteral stringTemplateLiteral, AnalyzerData
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral, AnalyzerData data) {
         // Check expr with insertions to resolve its type.
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:5832]: Check if this is related to IDL Client

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 10]
```diff
@@ -474,14 +472,10 @@ public BSymbol resolvePrefixSymbol(SymbolEnv env, Name pkgAlias, Name compUnit)
                 return symbol;
             }
 
-            if ((tag & SymTag.CLIENT_DECL) == SymTag.CLIENT_DECL) {
-                return resolveClientDeclPrefix(symbol);
-            }
-
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:480]: Removing this part only is sufficient

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/ResourceConfig.java [hunk 11]
```diff
@@ -64,11 +63,8 @@ public String name() {
         return name;
     }
 
-    public Optional<byte[]> content() {
-        if (content == null) {
-            return Optional.empty();
-        }
-        return Optional.of(content);
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/ResourceConfig.java:71]: Dont we need to keep these changes? @azinneera

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ResourceConfig.java:71]: The change was done for IDL client implementation. So reverting is fine.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java [hunk 12]
```diff
@@ -1618,7 +1611,8 @@ public void visit(BLangDynamicArgExpr dynamicParamExpr) {
     @Override
     public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         List<BLangExpression> interpolationsList =
-                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+                symResolver.getListOfInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList,
```
Comment (lasinicl) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java:1614]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java (hunk 1 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java:1466]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java (hunk 2 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:3412]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java (hunk 3 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java:2199]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java (hunk 4 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java:2016]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 13]
```diff
@@ -2623,8 +2596,8 @@ public BPackageSymbol getModuleForPackageId(PackageID packageID) {
                 .get();
     }
 
-    public List<BLangExpression> getListOfInterpolations(List<BLangExpression> sequenceList) {
-        List<BLangExpression> interpolationsList = new ArrayList<>();
+    public List<BLangExpression> getListOfInterpolations(List<BLangExpression> sequenceList,
+                                                          List<BLangExpression> interpolationsList) {
```
Comment (lasinicl) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:2600]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 14]
```diff
@@ -2642,7 +2615,8 @@ public List<BLangExpression> getListOfInterpolations(List<BLangExpression> seque
                 }
                 if (kind == NodeKind.REG_EXP_CAPTURING_GROUP) {
                     interpolationsList.addAll(
-                            getListOfInterpolations(((BLangReCapturingGroups) atom).disjunction.sequenceList));
+                            getListOfInterpolations(((BLangReCapturingGroups) atom).disjunction.sequenceList,
+                                    interpolationsList));
```
Comment (lasinicl) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:2619]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 9 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:5832]: This is related to reg exp.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/SymTag.java [hunk 15]
```diff
@@ -21,38 +21,37 @@
...
-    public static final long TUPLE_TYPE = 1 << 22 | TYPE_DEF;
-    public static final long ARRAY_TYPE = 1 << 23 | TYPE_DEF;
-    public static final long CONSTANT = 1 << 24 | VARIABLE_NAME | TYPE;
-    public static final long FUNCTION_TYPE = 1 << 25 | TYPE_DEF;
-    public static final long CONSTRUCTOR = 1 << 26 | INVOKABLE;
-    public static final long LET = 1 << 27;
-    public static final long ENUM = 1 << 28 | TYPE_DEF;
-    public static final long TYPE_REF = 1 << 29;
-    public static final long ANNOTATION_ATTACHMENT = 1 << 30;
-    public static final long CLIENT_DECL = 1 << 31 | IMPORT;
```
Comment (lasinicl) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/SymTag.java:56]: Shouldn't just this be removed?

Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/SymTag.java:56]: `CLIENT_DECL` is the only one removed in this file changes

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java (hunk 1 above)
Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ParameterDesugar.java]: Regex related changes are sorted.
please chec @gimantha @lasinicl

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BNeverType.java (hunk 5 above)
Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BNeverType.java]: Updated

File: compiler/ballerina-parser/src/test/resources/declarations/client-decl/client_decl_source_15.bal (hunk 6 above)
Comment (SasinduDilshara) [reply compiler/ballerina-parser/src/test/resources/declarations/client-decl/client_decl_source_15.bal]: Removed

File: language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation2.bal (hunk 8 above)
Comment (SasinduDilshara) [reply language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation2.bal]: Updated

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java (hunk 10 above)
Comment (SasinduDilshara) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:480]: Updated

File: language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation1.bal (hunk 7 above)
Comment (SasinduDilshara) [reply language-server/modules/langserver-core/src/test/resources/completion/annotation_ctx/source/moduleLevelAnnotation1.bal]: Updated

**Review by gimantha — APPROVED** [review]
LGTM

**Review by malinthar — APPROVED** [review]
